lib/fs/datablock_manager.cpp

```diff
@@ -0,0 +1,136 @@
+#include "fs.hpp"
+
+DataBlock_Manager::DataBlock_Manager(Fs *fs, u_int64_t block_segment_start,
+                                         u_int64_t block_segment_end)
+    : fs(fs), block_segment_start(block_segment_start),
+      block_segment_end(block_segment_end) {}
+
+class BitmapBlock_Data {
+public:
+  char buf[IO_BLOCK_SIZE];
+  u_int64_t datablocks_per_bitmap;

```

Comment:
This design is for eliminating the magic number caused bad branching? (like the i==253 stuff?)

Reply:
The datablocks_per_bitmap and implementation of find_unfilled is for increasing readability. 'find_unfilled' could possibly be speed up by x64 by using 64 bit operations instead of going one bit at a time. We also have the option of the last bitmap block on the disk tracking less datablocks than the others so that we don't waste any space on the disk.